catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js

```diff
@@ -14,332 +14,336 @@
-                            disableConfigs.each(function (disabledConfig) {
-                                if (disabledConfig.get('id') === content.id) {
-                                    toDelete.push(view.createDeletePromise(item, disabledConfig));
-                                }
-                            });
+                            var id = currentConfig ? currentConfig.get('id') : null;
+                            if (id === content.id) {
+                                toDelete.push(view.createDeletePromise(item, currentConfig));
+                            } else if (disableConfigs) {
+                                disableConfigs.each(function (disabledConfig) {
+                                    if (disabledConfig.get('id') === content.id) {
+                                        toDelete.push(view.createDeletePromise(item, disabledConfig));
+                                    }
+                                });
+                            }
                         }
-                    }
+                    });
                 });
-            });
 
-            //remove queued source configurations and entire source config if necessary
-            if (toDelete.length > 0) {
-                //remove all selected configurations from the current item
-                Q.all(toDelete).then(function(results) {
-                    _.each(results, function(result) {
-                        var item = result.source;
-                        if (item.size() <= 0) {
-                            //if no type configurations, delete the entire source.
-                            view.model.get('collection').removeSource(item);
-                            view.model.get('model').get('value').remove(item);
-                        }
+                //remove queued source configurations and entire source config if necessary
+                if (toDelete.length > 0) {
+                    //remove all selected configurations from the current item
+                    Q.all(toDelete).then(function (results) {
+                        _.each(results, function (result) {
+                            var item = result.source;
+                            if (item.size() <= 0) {
+                                //if no type configurations, delete the entire source.
+                                view.model.get('collection').removeSource(item);
+                                view.model.get('model').get('value').remove(item);
+                            }
+                        });
+                        wreqr.vent.trigger('refreshSources');
+                        view.$el.modal("hide");
+                    }).done();
+                }
+            },
+            createDeletePromise: function (source, config) {
+                var deferred = Q.defer();
+                var serviceModels = this.model.get('model').get('value');
+                config.destroy().success(function () {
```

Comment:
It doesn't look like this promise is chained correctly. `config.destroy()` will get called and then the function will return with the promise from Q, but the destroy promise will happen asynchronously from the Q promise. You could just return `config.destroy` I believe.